### PR TITLE
fix: start session from link bug

### PR DESCRIPTION
**Pull request by sjoerdlmkns**

### CODE REVIEW

**Review by sanderhollaar — APPROVED** [review]

File: lib/src/screens/home/home_screen.dart [hunk 1]
```diff
@@ -25,6 +29,40 @@ class _HomeScreenState extends State<HomeScreen> {
...
 
+  @override
+  void initState() {
+    super.initState();
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      final repo = IrmaRepositoryProvider.of(context);
+      final navigator = Navigator.of(context);
+
+      _maybeOpenQrScanner(repo, navigator);
```
Comment (ivard) [inline lib/src/screens/home/home_screen.dart:40]: I suspect that the QR scanner is not automatically opened on start-up now when the app becomes active from background.

File: lib/src/screens/home/widgets/pending_pointer_listener.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  @override
+  void initState() {
+    super.initState();
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      final repo = IrmaRepositoryProvider.of(context);
+      _pointerSubscription = repo.getPendingPointer().listen((Pointer? pointer) {
+        if (pointer != null) handlePointer(Navigator.of(context), pointer);
+      });
+    });
```
Comment (ivard) [inline lib/src/screens/home/widgets/pending_pointer_listener.dart:32]: Are we sure this also works when the app becomes active from background? I'd say that `initState` is not called then.

**Review by ivard — CHANGES_REQUESTED** [review]

File: lib/src/screens/home/widgets/pending_pointer_listener.dart (hunk 2 above)
Comment (sjoerdlmkns) [reply lib/src/screens/home/widgets/pending_pointer_listener.dart:32]: Yes this still works. The `PendingPointerListener` is wrapping the inner part of the `HomeScreen` which is always present in the stack (except during enrolment). `initState` only sets up the subscription and provides the callback so after that it just keeps listening to incoming pointers and executing the callback until the lifecycle of the widget ends.

File: lib/src/screens/home/home_screen.dart (hunk 1 above)
Comment (sjoerdlmkns) [reply lib/src/screens/home/home_screen.dart:40]: Yes this does not work anymore. To be honest i thought this was not expected behaviour so omitted it. But i will add it again when reactivating the app, so not just on launch. 👍🏻

**Review by ivard — APPROVED** [review]

**Review by MartijnYivi — APPROVED** [review]

**Pull request merged**